data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v2/beacon/PostAttestationsV2IntegrationTest.java

```diff
@@ -0,0 +1,185 @@
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.SpecMilestone;
+import tech.pegasys.teku.spec.TestSpecContext;
+import tech.pegasys.teku.spec.TestSpecInvocationContextProvider;
+import tech.pegasys.teku.spec.datastructures.operations.Attestation;
+import tech.pegasys.teku.spec.util.DataStructureUtil;
+import tech.pegasys.teku.validator.api.SubmitDataError;
+
+@TestSpecContext(milestone = {SpecMilestone.PHASE0, SpecMilestone.ELECTRA})
+public class PostAttestationsV2IntegrationTest extends AbstractDataBackedRestAPIIntegrationTest {
+
+  private DataStructureUtil dataStructureUtil;
+  private SpecMilestone specMilestone;
+
+  @BeforeEach
+  void setup(final TestSpecInvocationContextProvider.SpecContext specContext) {
+    spec = specContext.getSpec();
+    specMilestone = specContext.getSpecMilestone();
+    startRestAPIAtGenesis(specMilestone);
+    dataStructureUtil = specContext.getDataStructureUtil();
+  }
+
+  @TestTemplate
+  void shouldPostAttestations_NoErrors() throws Exception {
+    when(validatorApiChannel.sendSignedAttestations(anyList()))
+        .thenReturn(SafeFuture.completedFuture(Collections.emptyList()));
+
+    final List<Attestation> attestations =
+        List.of(dataStructureUtil.randomAttestation(), dataStructureUtil.randomAttestation());
+
+    final Response response =
+        post(
+            PostAttestationsV2.ROUTE,
+            JsonUtil.serialize(
+                attestations,
+                SerializableTypeDefinition.listOf(
+                    spec.atSlot(UInt64.ONE)
+                        .getSchemaDefinitions()
+                        .getAttestationSchema()
+                        .castTypeToAttestationSchema()
+                        .getJsonTypeDefinition())),
+            Collections.emptyMap(),
+            Optional.of(specMilestone.name().toLowerCase(Locale.ROOT)));
+
+    assertThat(response.code()).isEqualTo(SC_OK);
+    assertThat(response.body().string()).isEmpty();
+  }
+
+  @TestTemplate
+  void shouldPartiallyPostAttestations_ReturnsErrors() throws Exception {
+    final SubmitDataError firstSubmitDataError =
+        new SubmitDataError(UInt64.ZERO, "Bad attestation");
+    final SubmitDataError secondSubmitDataError =
+        new SubmitDataError(UInt64.ONE, "Very bad attestation");
+    when(validatorApiChannel.sendSignedAttestations(anyList()))
+        .thenReturn(
+            SafeFuture.completedFuture(List.of(firstSubmitDataError, secondSubmitDataError)));
+
+    final List<Attestation> attestations =
+        List.of(
+            dataStructureUtil.randomAttestation(),
+            dataStructureUtil.randomAttestation(),
+            dataStructureUtil.randomAttestation());
+
+    final Response response =
+        post(
+            PostAttestationsV2.ROUTE,
+            JsonUtil.serialize(
+                attestations,
+                SerializableTypeDefinition.listOf(
```

Comment:
nit: this is reused in all methods and could probably get extracted into a private method